naviqore_viewer/README.md

```diff
@@ -0,0 +1,22 @@
+# NAVIQORE Streamlit App
+
+## Setup
+
+### Install dependencies
+
+Run the command `poetry install` in your terminal to create a virtual environment with all requirements installed.
+
+### Configuration
+
+Create a `.env` file in the root directory. And add a line telling the application to which service host it should connect. In Local Development this is typically a Java Spring service running on `http://localhost:8080`.
+
+The required line in the `.env` file will then be: `NAVIQORE_HOST_URL=http://localhost:8080`
+
+_Note: you can also specify the `NAVIQORE_HOST_URL` in your operating system environment._
```

Comment:
The option with the env file worked, somehow exporting the env variable did not... ?

```sh
export NAVIQORE_HOST_URL=http://localhost:8080  
```

Error on startup:
```
ValueError: NAVIQORE_HOST_URL not found in .env file
Traceback:
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 600, in _run_script
    exec(code, module.__dict__)
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/naviqore_viewer/main.py", line 15, in <module>
    stops = getStops()
            ^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/caching/cache_utils.py", line 165, in wrapper
    return cached_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/caching/cache_utils.py", line 194, in __call__
    return self._get_or_create_cached_value(args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/caching/cache_utils.py", line 221, in _get_or_create_cached_value
    return self._handle_cache_miss(cache, value_key, func_args, func_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/caching/cache_utils.py", line 277, in _handle_cache_miss
    computed_value = self._info.func(*func_args, **func_kwargs)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/naviqore_viewer/client.py", line 38, in getStops
    client = getClient()
             ^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/caching/cache_utils.py", line 165, in wrapper
    return cached_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/caching/cache_utils.py", line 194, in __call__
    return self._get_or_create_cached_value(args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/caching/cache_utils.py", line 221, in _get_or_create_cached_value
    return self._handle_cache_miss(cache, value_key, func_args, func_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/.venv/lib/python3.12/site-packages/streamlit/runtime/caching/cache_utils.py", line 277, in _handle_cache_miss
    computed_value = self._info.func(*func_args, **func_kwargs)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/munterfi/repos/naviqore/public-transit-viewer/naviqore_viewer/naviqore_viewer/client.py", line 20, in getClient
    raise ValueError("NAVIQORE_HOST_URL not found in .env file")
```

Reply:
Interesting, I never tested but thought this must work according to the library documentation.


> **Variable expansion**
> 
> Python-dotenv can interpolate variables using POSIX variable expansion.
> 
> With load_dotenv(override=True) or dotenv_values(), the value of a variable is the first of the values defined in the following list:
> 
>     Value of that variable in the .env file.
>     Value of that variable in the environment.
>     Default value, if provided.
>     Empty string.

Maybe, I should revert back to coding this library myself instead of relying on third parties.

Reply:
Fixed in ca9857b